controllers/glance_controller.go

```diff
@@ -162,6 +162,21 @@ func (r *GlanceReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 			util.LogErrorForObject(helper, err, "Set after and calc patch/diff", instance)
 		}
 
+		if changed := helper.GetChanges()["metadata"]; changed {
+			// Patch wipes out all defaulted non-scalar fields from the spec (such as PasswordSelector)
+			// that were not explicitly included in the instance's CR YAML
+			// if err := r.Patch(ctx, instance, patch); err != nil && !k8s_errors.IsNotFound(err) {
+			// 	util.LogErrorForObject(helper, err, "Update finalizer", instance)
+			// }
+
+			// Update does not exhibit the "wipes out defaulted non-scalar fields" behavior mentioned above
+			if err := r.Update(ctx, instance); err != nil && !k8s_errors.IsNotFound(err) {
+				util.LogErrorForObject(helper, err, "Update metadata", instance)
+			}
+		}
+
+		// Calling "r.Status().Patch()" BEFORE handling the metadata causes any metadata changes to be
```

Comment:
This is now AFTER the metadata change

Reply:
Right, that's why it's after instead of before (and I just wanted to explain why I put it after instead of before).  Let me clarify the comment.